### PR TITLE
Use encrypter.Interface for encryption in s3object and cloudconfig

### PR DESCRIPTION
**Pull request by fgimenez**
All KMS functionality replaced by the injected encrypter. Only missing the decryption part, this will be changed on the templates once we can select different encryption backends.

### CODE REVIEW

File: service/controller/v13/cloudconfig/cloud_config_test.go [hunk 1]
```diff
@@ -3,13 +3,14 @@ package cloudconfig
...
 	"compress/gzip"
+	"context"
 	"encoding/base64"
 	"io"
 	"strings"
 	"testing"
 
-	"github.com/aws/aws-sdk-go/service/kms"
 	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/aws-operator/service/controller/v13/encrypter"
```
Comment (xh3b4sd) [inline service/controller/v13/cloudconfig/cloud_config_test.go]: Would be cool to have the operator import in its own separate block. Same is above.

File: service/controller/v13/cloudconfig/interface.go [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+
 	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/certs/legacy"
 	"github.com/giantswarm/randomkeys"
 )
 
 type Interface interface {
-	NewMasterTemplate(customObject v1alpha1.AWSConfig, certs legacy.CompactTLSAssets, clusterKeys randomkeys.Cluster, kmsKeyARN string) (string, error)
+	NewMasterTemplate(ctx context.Context, customObject v1alpha1.AWSConfig, certs legacy.CompactTLSAssets, clusterKeys randomkeys.Cluster) (string, error)
 	NewWorkerTemplate(customObject v1alpha1.AWSConfig, certs legacy.CompactTLSAssets) (string, error)
```
Comment (xh3b4sd) [inline service/controller/v13/cloudconfig/interface.go]: My OCD thinks we should provide a context here as well. 🙈

File: service/controller/v13/encrypter/kms/certs.go [hunk 3]
```diff
@@ -156,3 +167,29 @@ func assetsBundleKey(c legacy.ClusterComponent, t legacy.TLSAssetType) legacy.As
...
+func encryptor(svc KMSClient, kmsKeyARN string, data []byte) ([]byte, error) {
+	encryptInput := kms.EncryptInput{
+		KeyId:     aws.String(kmsKeyARN),
+		Plaintext: data,
+	}
+
+	var encryptOutput *kms.EncryptOutput
+	var err error
+	if encryptOutput, err = svc.Encrypt(&encryptInput); err != nil {
+		return []byte{}, err
```
Comment (xh3b4sd) [inline service/controller/v13/encrypter/kms/certs.go]: Zero value of a slice is `nil`.

File: service/controller/v13/encrypter/spec.go [hunk 4]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 
 	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/certs/legacy"
```
Comment (xh3b4sd) [inline service/controller/v13/encrypter/spec.go]: I noticed this when we worked on the k8s libs updates as well. We still use some legacy crap here. Would be cool to work towards cleaning this up a bit in the future.

File: service/controller/v13/resource/encryptionkey/delete_test.go [hunk 5]
```diff
@@ -5,9 +5,8 @@ import (
 	"testing"
 
 	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
-	"github.com/giantswarm/micrologger/microloggertest"
-
 	"github.com/giantswarm/aws-operator/service/controller/v13/encrypter"
+	"github.com/giantswarm/micrologger/microloggertest"
```
Comment (xh3b4sd) [inline service/controller/v13/resource/encryptionkey/delete_test.go]: The blocks are weird here as well. Below as well.

File: service/controller/v13/resource/s3object/desired.go [hunk 6]
```diff
@@ -42,7 +42,7 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		return output, microerror.Mask(err)
 	}
 
-	tlsAssets, err := r.encodeTLSAssets(ctx, certs, kmsKeyARN)
+	tlsAssets, err := r.encrypter.EncryptTLSAssets(ctx, customObject, certs)
```
Comment (xh3b4sd) [inline service/controller/v13/resource/s3object/desired.go]: This is really cool that we work on this abstraction here now. The project improves that way quite nicely.

File: service/controller/v13/cloudconfig/cloud_config_test.go (hunk 1 above)
Comment (fgimenez) [reply service/controller/v13/cloudconfig/cloud_config_test.go]: Sure, done.

File: service/controller/v13/cloudconfig/interface.go (hunk 2 above)
Comment (fgimenez) [reply service/controller/v13/cloudconfig/interface.go]: Done, not needed right now but good to have it in place.

File: service/controller/v13/encrypter/kms/certs.go (hunk 3 above)
Comment (fgimenez) [reply service/controller/v13/encrypter/kms/certs.go]: Sure, fixed.

File: service/controller/v13/resource/encryptionkey/delete_test.go (hunk 5 above)
Comment (fgimenez) [reply service/controller/v13/resource/encryptionkey/delete_test.go]: Indeed, fixed.

File: service/controller/v13/encrypter/spec.go (hunk 4 above)
Comment (rossf7) [reply service/controller/v13/encrypter/spec.go]: AIUI we can't have both the new certs and the legacy package vendored.

The legacy certs are used in the v1 (legacy) resource. We didn't want to invest time unpicking it there. So this is waiting on the cluster upgrades until we can delete v1.

It sucks but I've tried upgrading previously and got blocked by this.

Comment (xh3b4sd) [reply service/controller/v13/encrypter/spec.go]: I see. Makes sense.

**Comment by fgimenez** [comment]
Tested on gauss

**Pull request merged**